entrypoint.sh

```diff
@@ -100,37 +99,131 @@ function super_lint() {
   /action/lib/linter.sh
 }
 
+function get_files_matching_filetype() {
+  local matching_files=()
+  local filenames=("${@:2}")
+  local filetype=$1
+
+  for file in "${filenames[@]}"; do
+    filename=$(basename "$file")
+
+    if [[ "$filename" == *"$filetype" ]]; then
+      matching_files+=("$file")
+    fi
+  done
+  echo "${matching_files[@]}"
+}
+
+function linter_failed() {
+  return=$1
+
+  if [[ "${return:-1}" != 0 ]]; then
+    linter_exit_codes+=(["${linter[name]}"]="${return}")
+  else
+    return "${return}"
+  fi
+}
+
+function lint_files() {
+  exec &>> "${linter[logfile]}"
+  if [[ "${linter[filetype]}" = "all" ]]; then
+    cmd="${linter[name]} ${linter[args]}"
+    eval "$cmd"
+    linter_failed $?
+  else
+    for file in $(get_files_matching_filetype "${linter[filetype]}" "${included[@]}"); do 
+      if [[ "${linter[executor]+x}" ]]; then
+        cmd="${linter[executor]} -c \"${linter[name]} ${linter[args]} ${file}\""
+      else
+        cmd="${linter[name]} ${linter[args]} ${file}"
+      fi
+      eval "$cmd"
+      linter_failed $?
+    done
+  fi
+}
+
 function seiso_lint() {
+  echo -e "\nRunning Seiso Linter\n--------------------------\n"
+
   excluded=()
   included=()
 
   while read -r file; do
-    # Apply filter with =~ to ensure it is aligned with github/super-linter
     if [[ -n ${INPUT_EXCLUDE:+x} && "${file}" =~ ${INPUT_EXCLUDE} ]]; then
       excluded+=("${file}")
       continue
     fi
-
     included+=("${file}")
+  done < <(find . -path "./.git" -prune -or -type f)
+  
+  declare -gA linter_exit_codes
+  input="/etc/opt/goat/linters.txt"
 
-    # Check Dockerfiles
-    if [[ "${file}" = *Dockerfile ]]; then
-      dockerfile_lint -f "${file}" -r /etc/opt/goat/oci.yml
+  while read -r line; do
+    if [[ $line == \#* ]]; then
+      continue
     fi
 
-    # Check .md file spelling and links
-    if [[ "${file}" = *.md ]]; then
-      npx cspell -c /etc/opt/goat/cspell.config.js -- "${file}"
-      npx markdown-link-check --config /etc/opt/goat/links.json --verbose "${file}"
+    IFS="," read -ra pairs <<< "$line"
+ 
+    unset linter
+    declare -gA linter
+
+    for pair in "${pairs[@]}"; do
```

Comment:
Clear the linter array for each line in linters.txt and read in the values for the current run